### PR TITLE
feat: add support for ephemeral storage requests and limits to casc t…

### LINKED ISSUE #935: Support EphemeralStorage requests and limits in CASC templates
### Is your feature request related to a problem? Please describe

I'd like to be able to set EphemeralStorage requests and limits in the resources section of the casc template

### Describe the solution you'd like

Similar to requestMemory and requestCPU I'd like requestEphemeralStorage and limitEphemeralStorage to be set

### Describe alternatives you've considered

_No response_

### Additional context

https://www.jenkins.io/doc/pipeline/steps/kubernetes/ supports ephemeral storage

### PR DESCRIPTION
**Pull request by knechtionscoding**
…emplates

### What does this PR do?

- Fixes #935 

If you modified files in the `./charts/jenkins/` directory, please also include the following:

```[tasklist]
### Submitter checklist
- [x] I bumped the "version" key in `./charts/jenkins/Chart.yml`.
- [x] I added a new changelog entry to `./charts/jenkins/CHANGELOG.md`.
- [x] I followed the [technical requirements](https://github.com/jenkinsci/helm-charts/blob/main/CONTRIBUTING.md#technical-requirements).
```

### CODE REVIEW

**Comment by knechtionscoding** [comment]
```bash
hknecht:helm-charts-jenkins/ (main✗) $ helm unittest --helm3 --strict -f 'unittests/*.yaml' charts/jenkins                                                                                              

### Chart [ jenkins ] charts/jenkins

2023/10/18 21:21:03 warning: destination for jenkins.controller.initScripts is a table. Ignoring non-table value ([])
 PASS  ConfigMap        charts/jenkins/unittests/config-init-scripts-test.yaml
 PASS  ConfigMap        charts/jenkins/unittests/config-test.yaml
 PASS  PersistentVolumeClaim    charts/jenkins/unittests/home-pvc-test.yaml
2023/10/18 21:21:03 warning: destination for jenkins.controller.projectNamingStrategy is a table. Ignoring non-table value (standard)
2023/10/18 21:21:03 warning: cannot overwrite table with non table for jenkins.controller.JCasC.security.apiToken (map[creationOfLegacyTokenEnabled:false tokenGenerationOnCreationEnabled:false usageStatisticsEnabled:true])
 PASS  Configuration as Code    charts/jenkins/unittests/jcasc-config-test.yaml
 PASS  Jenkins Agent Service    charts/jenkins/unittests/jenkins-agent-svc-test.yaml
 PASS  Jenkins Backup Cronjob   charts/jenkins/unittests/jenkins-backup-cronjob-test.yaml
 PASS  Controller Prometheus PrometheusRule     charts/jenkins/unittests/jenkins-controller-alerting-rules-test.yaml
 PASS  Controller Primary Ingress       charts/jenkins/unittests/jenkins-controller-ingress-1.19-test.yaml
 PASS  Controller Ingress - DRY Support charts/jenkins/unittests/jenkins-controller-ingress-DRY-test.yaml
 PASS  Controller Primary Ingress       charts/jenkins/unittests/jenkins-controller-ingress-test.yaml
 PASS  Network Policy   charts/jenkins/unittests/jenkins-controller-networkpolicy-test.yaml
 PASS  Controller Pod Disruption Budget charts/jenkins/unittests/jenkins-controller-pdb-1.21-test.yaml
 PASS  Controller Pod Disruption Budget charts/jenkins/unittests/jenkins-controller-pdb-test.yaml
 PASS  Controller Secondary Ingress     charts/jenkins/unittests/jenkins-controller-secondary-ingress-1.19-test.yaml
 PASS  Controller Secondary Ingress     charts/jenkins/unittests/jenkins-controller-secondary-ingress-test.yaml
 PASS  Controller Prometheus ServiceMonitor     charts/jenkins/unittests/jenkins-controller-servicemonitor_test.yaml
 PASS  Jenkins Controller       charts/jenkins/unittests/jenkins-controller-statefulset-test.yaml
 PASS  Jenkins Controller       charts/jenkins/unittests/jenkins-controller-svc-test.yaml
 PASS  Role Based Access Control        charts/jenkins/unittests/rbac-test.yaml
 PASS  Controller Admin Additional Secrets      charts/jenkins/unittests/secret-additional-test.yaml
 PASS  Controller Secret Claims charts/jenkins/unittests/secret-claims-test.yaml
 PASS  Controller Additional Existing Secrets   charts/jenkins/unittests/secret-existing-test.yaml
 PASS  Controller Admin Credentials     charts/jenkins/unittests/secret-test.yaml
 PASS  Controller Service Account       charts/jenkins/unittests/service-account-agent-test.yaml
 PASS  Controller Service Account       charts/jenkins/unittests/service-account-test.yaml

Charts:      1 passed, 1 total
Test Suites: 25 passed, 25 total
Tests:       154 passed, 154 total
Snapshot:    1 passed, 1 total
Time:        1.00814175s
```

It passes linting locally. I'm not sure why the linting for 1.26.4 is failing. It appears to be failing on the download, so I think it's just a flake?

File: charts/jenkins/values.yaml [hunk 1]
```diff
@@ -651,9 +651,11 @@ agent:
     requests:
       cpu: "512m"
       memory: "512Mi"
+      ephemeralStorage: ""
```
Comment (timja) [inline charts/jenkins/values.yaml:654]:
```suggestion
      # ephemeralStorage:
```

File: charts/jenkins/values.yaml [hunk 2]
```diff
@@ -651,9 +651,11 @@ agent:
...
     limits:
       cpu: "512m"
       memory: "512Mi"
+      ephemeralStorage: ""
```
Comment (timja) [inline charts/jenkins/values.yaml:658]:
```suggestion
      # ephemeralStorage:
```

File: charts/jenkins/CHANGELOG.md [hunk 3]
```diff
@@ -12,6 +12,10 @@ Use the following links to reference issues, PRs, and commits prior to v2.6.0.
 The changelog until v1.5.7 was auto-generated based on git commits.
 Those entries include a reference to the git commit to be able to get more details.
 
+## 4.8.0
+
+Adds support for ephemeralStorage in CASC Templates
```
Comment (timja) [inline charts/jenkins/CHANGELOG.md:17]: what about this?
```suggestion
Adds support for ephemeralStorage request and limit in Kubernetes plugin JCasC template
```

File: charts/jenkins/values.yaml (hunk 1 above)
Comment (knechtionscoding) [reply charts/jenkins/values.yaml:654]: Sure, the "" value is null is valid to pass in, and the conditional should stop it from being put into the cluster/plugin unless you actually insert a value.

I don't really mind one way or the other.

**Review by timja — APPROVED** [review]
Thanks!

**Pull request merged**